packages/flutter/lib/src/widgets/animated_switcher.dart

```diff
@@ -183,82 +188,97 @@ class AnimatedSwitcher extends StatefulWidget {
   @override
   _AnimatedSwitcherState createState() => new _AnimatedSwitcherState();
 
-  /// The default transition algorithm used by [AnimatedSwitcher].
+  /// The transition builder used as the default value of [transitionBuilder].
   ///
   /// The new child is given a [FadeTransition] which increases opacity as
   /// the animation goes from 0.0 to 1.0, and decreases when the animation is
   /// reversed.
   ///
-  /// The default value for the [transitionBuilder], an
-  /// [AnimatedSwitcherTransitionBuilder] function.
+  /// This is an [AnimatedSwitcherTransitionBuilder] function.
   static Widget defaultTransitionBuilder(Widget child, Animation<double> animation) {
     return new FadeTransition(
       opacity: animation,
       child: child,
     );
   }
 
-  /// The default layout algorithm used by [AnimatedSwitcher].
+  /// The layout builder used as the default value of [layoutBuilder].
   ///
   /// The new child is placed in a [Stack] that sizes itself to match the
   /// largest of the child or a previous child. The children are centered on
   /// each other.
   ///
-  /// This is the default value for [layoutBuilder]. It implements
-  /// [AnimatedSwitcherLayoutBuilder].
-  static Widget defaultLayoutBuilder(List<Widget> children) {
+  /// This is an [AnimatedSwitcherLayoutBuilder] function.
+  static Widget defaultLayoutBuilder(Widget currentChild, List<Widget> previousChildren) {
     return new Stack(
-      children: children,
+      children: currentChild == null ? previousChildren : previousChildren + <Widget>[currentChild],
```

Comment:
allocating three lists each time looks expensive...

Reply:
How about:

```dart
List<Widget> children = previousChildren;
if (currentChild != null)
  children = children.toList()..add(currentChild);
```

...then pass `children`.

Reply:
OK, that works.  I inlined it though.

```dart
currentChild == null ? previousChildren : previousChildren.toList()..add(currentChild)
```